refactor(deviation_evaluator): simplify plot function

kminoda opened the pull request:
Signed-off-by: kminoda <koji.minoda@tier4.jp>

## Description
I refactored deviation_evaluation_visualizer script for better readability.
The performance does not change before and after this PR except:
- `lower_bound` calculation (fixed a small mistake in previous version)
- duration-to-error plot (removed the plot, since it seemed unnecessary)

<!-- Write a brief description of this PR. -->

## Related links

<!-- Write the links related to this PR. -->

## Tests performed
Ran the tool for several data and confirmed that the results are valid
<!-- Describe how you have tested this PR. -->

## Notes for reviewers

<!-- Write additional information if necessary. It should be written if there are related PRs that should be merged at the same time. -->

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

kminoda committed:
refactor(deviation_evaluator): simplify plot function

Signed-off-by: kminoda <koji.minoda@tier4.jp>

pre-commit-ci[bot] committed:
ci(pre-commit): autofix

kminoda committed:
simplified more

Signed-off-by: kminoda <koji.minoda@tier4.jp>

kminoda committed:
resolve conflicts

Signed-off-by: kminoda <koji.minoda@tier4.jp>

kminoda commented:
Is it OK for me to request you a review for the modification in the python script...?

kminoda committed:
Merge remote-tracking branch 'origin' into refactor/deviation_evaluation_visualizer/simplify_plot

kminoda committed:
works?

Signed-off-by: kminoda <koji.minoda@tier4.jp>

pre-commit-ci[bot] committed:
ci(pre-commit): autofix

kminoda committed:
refactoring finished

Signed-off-by: kminoda <koji.minoda@tier4.jp>

kminoda committed:
resolved conflict

Signed-off-by: kminoda <koji.minoda@tier4.jp>

kminoda committed:
resolve conflict

Signed-off-by: kminoda <koji.minoda@tier4.jp>

kminoda committed:
minor fix

Signed-off-by: kminoda <koji.minoda@tier4.jp>

kminoda committed:
debugged

Signed-off-by: kminoda <koji.minoda@tier4.jp>

kminoda committed:
refactored

Signed-off-by: kminoda <koji.m.minoda@gmail.com>

pre-commit-ci[bot] committed:
ci(pre-commit): autofix

kminoda committed:
refactored recall calculation

Signed-off-by: kminoda <koji.m.minoda@gmail.com>

kminoda committed:
reverted thread library import

Signed-off-by: kminoda <koji.m.minoda@gmail.com>

kminoda committed:
fix param file to default ones

Signed-off-by: kminoda <koji.minoda@tier4.jp>

kminoda commented:
@YamatoAndo こちらお時間のあるときにお願いします m(_ _)m

YamatoAndo commented on localization/deviation_estimation_tools/deviation_evaluator/scripts/bag_load_utils.py:
```diff
@@ -0,0 +1,336 @@
+        pose_list = np.array(pose_list)[sorted_idxs]
+        pose_cov_list = np.array(pose_cov_list)[sorted_idxs]
+        pose_list = np.array(pose_list)
+        pose_cov_list = np.array(pose_cov_list)
```
unnecessary lines

kminoda replied:
Let me resolve this in the following PRs

YamatoAndo commented on localization/deviation_estimation_tools/deviation_evaluator/scripts/bag_load_utils.py:
```diff
@@ -0,0 +1,336 @@
+        pose_list = np.array(pose_list)[sorted_idxs]
+        pose_cov_list = np.array(pose_cov_list)[sorted_idxs]
+        pose_list = np.array(pose_list)
+        pose_cov_list = np.array(pose_cov_list)
```
unnecessary lines

kminoda committed:
dummy commit

pre-commit-ci[bot] committed:
ci(pre-commit): autofix

kminoda commented:
@kenji-miyake Sorry to bother you.
`pre-commit` GitHub Actions does not run somehow (it's skipped). Do you have any idea how to make it run so that this PR can be merged?

kenji-miyake commented:
@kminoda It has run, but failed. Please fix them manually since some fixes aren't automatically applied.

kminoda commented:
I see, thank you for the reply!

kminoda committed:
fix pre-commit

Signed-off-by: kminoda <koji.minoda@tier4.jp>

pre-commit-ci[bot] committed:
ci(pre-commit): autofix

kminoda committed:
reflected comment

Signed-off-by: kminoda <koji.minoda@tier4.jp>

kminoda committed:
Merge branch 'refactor/deviation_evaluation_visualizer/simplify_plot' of github.com:tier4/CalibrationTools into refactor/deviation_evaluation_visualizer/simplify_plot

kminoda committed:
fix pre-commit

Signed-off-by: kminoda <koji.minoda@tier4.jp>

kminoda committed:
use static_cast

Signed-off-by: kminoda <koji.minoda@tier4.jp>

pre-commit-ci[bot] committed:
ci(pre-commit): autofix

kminoda commented:
@kenji-miyake I fixed the pre-commit warnings on the scripts I modified, but still the `pre-commit / pre-commit (pull_request)` is skipped. Do you know how to fix this?
![Screenshot from 2022-11-01 18-13-38](https://user-images.githubusercontent.com/44218668/199200480-8b307d02-8dfa-4ab4-81ad-37b9b7b8b9b2.png)

kenji-miyake commented:
@kminoda Ah, I see. It's intended to run only on private repositories.
For public repositories, please see `pre-commit.ci - pr` instead.

kminoda commented:
@kenji-miyake OK! But another problem is that the `pre-commit / pre-commit (pull_request)` is set as a required workflow for merging the PR. Maybe we should set this as a non-mandatory workflow?

kenji-miyake commented:
Exactly.
@yabuta Could you confirm the settings? :pray:

yabuta commented:
@kminoda @kenji-miyake I changed this setting.

kminoda committed:
Merge branch 'tier4/universe' into refactor/deviation_evaluation_visualizer/simplify_plot
